eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -149,63 +149,49 @@ sgActiveSlotCoeff :: ShelleyGenesis era -> ActiveSlotCoeff
 sgActiveSlotCoeff = mkActiveSlotCoeff . sgActiveSlotsCoeff
 
 instance Era era => ToJSON (ShelleyGenesis era) where
-  toJSON sg =
-    Aeson.object
-      [ "systemStart" .= sgSystemStart sg,
-        "networkMagic" .= sgNetworkMagic sg,
-        "networkId" .= sgNetworkId sg,
-        "activeSlotsCoeff" .= sgActiveSlotsCoeff sg,
-        "securityParam" .= sgSecurityParam sg,
-        "epochLength" .= sgEpochLength sg,
-        "slotsPerKESPeriod" .= sgSlotsPerKESPeriod sg,
-        "maxKESEvolutions" .= sgMaxKESEvolutions sg,
-        "slotLength" .= sgSlotLength sg,
-        "updateQuorum" .= sgUpdateQuorum sg,
-        "maxLovelaceSupply" .= sgMaxLovelaceSupply sg,
-        "protocolParams" .= sgProtocolParams sg,
-        "genDelegs" .= sgGenDelegs sg,
-        "initialFunds" .= sgInitialFunds sg,
-        "staking" .= sgStaking sg
-      ]
+  toJSON = Aeson.object . toShelleyGenesisPairs
+  toEncoding = Aeson.pairs . mconcat . toShelleyGenesisPairs
 
-  toEncoding
-    ShelleyGenesis
-      { sgSystemStart,
-        sgNetworkMagic,
-        sgNetworkId,
-        sgActiveSlotsCoeff,
-        sgSecurityParam,
-        sgEpochLength,
-        sgSlotsPerKESPeriod,
-        sgMaxKESEvolutions,
-        sgSlotLength,
-        sgUpdateQuorum,
-        sgMaxLovelaceSupply,
-        sgProtocolParams,
-        sgGenDelegs,
-        sgInitialFunds,
-        sgStaking
-      } =
-      Aeson.pairs $
-        let !strictSgInitialFunds = sgInitialFunds
-            !strictSgStaking = sgStaking
-         in mconcat
-              [ "systemStart" .= sgSystemStart,
-                "networkMagic" .= sgNetworkMagic,
-                "networkId" .= sgNetworkId,
-                "activeSlotsCoeff" .= sgActiveSlotsCoeff,
-                "securityParam" .= sgSecurityParam,
-                "epochLength" .= sgEpochLength,
-                "slotsPerKESPeriod" .= sgSlotsPerKESPeriod,
-                "maxKESEvolutions" .= sgMaxKESEvolutions,
-                "slotLength" .= sgSlotLength,
-                "updateQuorum" .= sgUpdateQuorum,
-                "maxLovelaceSupply" .= sgMaxLovelaceSupply,
-                "protocolParams" .= sgProtocolParams,
-                "genDelegs" .= sgGenDelegs,
-                "initialFunds" .= strictSgInitialFunds,
-                "staking" .= strictSgStaking
-              ]
+toShelleyGenesisPairs ::
+  (Aeson.KeyValue a, CC.Crypto (Crypto era)) =>
+  ShelleyGenesis era ->
+  [a]
+toShelleyGenesisPairs
+  ShelleyGenesis
+    { sgSystemStart,
+      sgNetworkMagic,
+      sgNetworkId,
+      sgActiveSlotsCoeff,
+      sgSecurityParam,
+      sgEpochLength,
+      sgSlotsPerKESPeriod,
+      sgMaxKESEvolutions,
+      sgSlotLength,
+      sgUpdateQuorum,
+      sgMaxLovelaceSupply,
+      sgProtocolParams,
+      sgGenDelegs,
+      sgInitialFunds,
+      sgStaking
+    } =
+    let !strictSgInitialFunds = sgInitialFunds
+        !strictSgStaking = sgStaking
```

Comment:
The fact that these two fields will be strict in `toJSON` after this change has no impact, correct?

Reply:
Yeah, it has no impact on the `toEncoding`, but for the `toJSON` it actually makes it is better.